lightning/src/ln/channelmanager.rs

```diff
@@ -4404,3 +4404,150 @@ mod tests {
 		}
 	}
 }
+
+#[cfg(all(test, feature = "unstable"))]
+mod benches {
+	use chain::Listen;
+	use chain::chainmonitor::ChainMonitor;
+	use chain::keysinterface::{KeysManager, InMemorySigner};
+	use chain::transaction::OutPoint;
+	use ln::channelmanager::{ChainParameters, ChannelManager, PaymentHash, PaymentPreimage};
+	use ln::features::InitFeatures;
+	use ln::functional_test_utils::*;
+	use ln::msgs::ChannelMessageHandler;
+	use routing::network_graph::NetworkGraph;
+	use routing::router::get_route;
+	use util::test_utils;
+	use util::config::UserConfig;
+	use util::events::{Event, EventsProvider, MessageSendEvent, MessageSendEventsProvider};
+
+	use bitcoin::hashes::Hash;
+	use bitcoin::hashes::sha256::Hash as Sha256;
+	use bitcoin::{Block, BlockHeader, Transaction, TxOut};
+
+	use std::sync::Mutex;
+
+	use test::Bencher;
+
+	struct NodeHolder<'a> {
+		node: &'a ChannelManager<InMemorySigner,
+			&'a ChainMonitor<InMemorySigner, &'a test_utils::TestChainSource,
+				&'a test_utils::TestBroadcaster, &'a test_utils::TestFeeEstimator,
+				&'a test_utils::TestLogger, &'a test_utils::TestPersister>,
+			&'a test_utils::TestBroadcaster, &'a KeysManager,
+			&'a test_utils::TestFeeEstimator, &'a test_utils::TestLogger>
+	}
+
+	#[bench]
+	fn bench_sends(bench: &mut Bencher) {
```

Comment:
Haven't seen `bench` tests before, is the idea that for each relevant PR we'd check this test to make sure payment latency wasn't impacted too much?

Reply:
I dunno that we care about monitoring it that closely, but it will be run on CI.